### PR TITLE
Detection eval max boxes fix

### PR DESCRIPTION
**Pull request by benjaminrwilson**
This is a followup PR. This adds two small things:

1. Unit test for `rank`.
2. A small revision previously discussed to remove the conditional in `rank`.

### CODE REVIEW

File: tests/test_eval_detection.py [hunk 1]
```diff
@@ -215,8 +216,26 @@ def test_accumulate() -> None:
         cls_to_accum["VEHICLE"]
         == np.array(
             [
-                [1.0, 1.0, 1.0, 1.0, expected_ATE, expected_ASE, expected_AOE, expected_AP],
-                [1.0, 1.0, 1.0, 1.0, expected_ATE, expected_ASE, expected_AOE, expected_AP],
+                [
```
Comment (johnwlambert) [inline tests/test_eval_detection.py:219]: i think it was more readable before, on a single line. If we do

```
# fmt: off

blah blah 

# fmt: on
blah blah
```
we can trick python black into keeping it as before

File: tests/test_eval_detection.py [hunk 2]
```diff
@@ -420,14 +439,23 @@ def test_remove_duplicate_instances_ground_truth():
...
-    """ Use the map to filter out an object that lies outside the ROI in a parking lot """
+def test_filter_objs_to_roi() -> None:
+    """Use the map to filter out an object that lies outside the ROI in a parking lot."""
     avm = ArgoverseMap()
 
     # should be outside of ROI
     outside_obj = {
-        "center": {"x": -14.102872067388489, "y": 19.466695178746022, "z": 0.11740010190455852},
-        "rotation": {"x": 0.0, "y": 0.0, "z": -0.038991328555453404, "w": 0.9992395490058831},
+        "center": {
```
Comment (johnwlambert) [inline tests/test_eval_detection.py:448]: any idea why these got reformatted? did you use `black myfile -l 120`?
do you think it is clearer on 1 line, or multiple lines? I think clearer on one line

File: tests/test_eval_detection.py [hunk 3]
```diff
@@ -474,3 +511,47 @@ def test_AP_on_filtered_instances() -> None:
     metrics = evaluator.evaluate()
 
     assert metrics.AP.loc["Vehicle"] == 1.0
+
+
+def test_rank() -> None:
```
Comment (johnwlambert) [inline tests/test_eval_detection.py:516]: looks good, thanks for adding this

File: tests/test_eval_detection.py (hunk 2 above)
Comment (benjaminrwilson) [reply tests/test_eval_detection.py:448]: I'll check on this. I opened: [PR-228](https://github.com/argoai/argoverse-api/pull/228) to hopefully fix formatting related stuff moving forward.

Comment (benjaminrwilson) [reply tests/test_eval_detection.py:448]: Okay, I reran black. All the files should be updated now.

File: argoverse/utils/transform.py [hunk 4]
```diff
@@ -49,10 +49,10 @@ def quat_argo2scipy(q: np.ndarray) -> np.ndarray:
 
 
 def quat_argo2scipy_vectorized(q: np.ndarray) -> np.ndarray:
-    """"Re-order Argoverse's scalar-first [w,x,y,z] quaternion order to Scipy's scalar-last [x,y,z,w]"""
+    """ "Re-order Argoverse's scalar-first [w,x,y,z] quaternion order to Scipy's scalar-last [x,y,z,w]"""
```
Comment (jagjeet-singh) [inline argoverse/utils/transform.py:52]: Formatting seems weird. 3 Quotes followed by a space and another quote.

File: demo_usage/visualize_30hz_benchmark_data_on_map.py [hunk 5]
```diff
@@ -327,7 +327,7 @@ def render_front_camera_on_axis(self, ax: plt.Axes, timestamp: int, log_id: str)
 
 
 def visualize_30hz_benchmark_data_on_map(args: Any) -> None:
-    """"""
+    """ """
```
Comment (jagjeet-singh) [inline demo_usage/visualize_30hz_benchmark_data_on_map.py:330]: docstring missing?

File: integration_tests/test_map_api.py [hunk 6]
```diff
@@ -25,14 +25,14 @@ def add_lane_segment_to_ax(
     ymin: float,
     ymax: float,
 ) -> None:
-    """"""
+    """ """
```
Comment (jagjeet-singh) [inline integration_tests/test_map_api.py:28]: same

Comment (jagjeet-singh) [reply integration_tests/test_map_api.py:28]: and in other functions too

File: tests/test_manhattan_search.py [hunk 7]
```diff
@@ -75,7 +75,7 @@ def polygons_and_gt_bboxes() -> Tuple[List[np.ndarray], List[np.ndarray]]:
 def test_find_all_polygon_bboxes_overlapping_query_bbox(
     polygons_and_gt_bboxes: Tuple[List[np.ndarray], List[np.ndarray]]
 ) -> None:
-    """Test for correctness of """
+    """Test for correctness of"""
```
Comment (jagjeet-singh) [inline tests/test_manhattan_search.py:78]: docstring incomplete?

**Review by jagjeet-singh — CHANGES_REQUESTED** [review]
Left some minor comments

File: integration_tests/test_map_api.py [hunk 8]
```diff
@@ -25,14 +25,14 @@ def add_lane_segment_to_ax(
...
-    """"""
+    """Utility function to add a lane segment to an Matplotlib axis."""
     plot_lane_segment_patch(lane_polygon, ax, color=patch_color, alpha=0.3)
 
 
 def find_lane_segment_bounds_in_table(
     avm: ArgoverseMap, city_name: str, lane_segment_id: int
 ) -> Tuple[float, float, float, float]:
-    """"""
+    """ """
```
Comment (tagarwal-argoai) [inline integration_tests/test_map_api.py:35]: Why are all of these docstring left empty? Is that intentional?

File: tests/test_eval_detection.py [hunk 9]
```diff
@@ -474,3 +511,47 @@ def test_AP_on_filtered_instances() -> None:
...
+                track_id="2",
+            ),
+        ]
+    )
+
+    ranked_dts, ranked_scores = rank(dts)
+    track_ids = np.array([dt.track_id for dt in ranked_dts.tolist()])
+    expected_track_ids = np.array(["1", "2", "0"])
+    expected_scores = np.array([0.9, 0.8, 0.7])
+    assert (track_ids == expected_track_ids).all() and (ranked_scores == expected_scores).all()
```
Comment (tagarwal-argoai) [inline tests/test_eval_detection.py:557]: [nit]: Just a readability comment: How do you feel like doing?
`all((track_ids == expected_track_ids))` as that is more readable than what we have currently.

File: tests/test_ffmpeg_utils_unit.py [hunk 10]
```diff
@@ -5,14 +5,14 @@
 
 
 def test_ffmpeg_seq_frame_vid_smokescreen() -> None:
-    """"""
+    """ """
```
Comment (tagarwal-argoai) [inline tests/test_ffmpeg_utils_unit.py:8]: Same comment on empty docstrings here and below as well? Do you think we can add a quick 1-line docstring for them?

**Review by tagarwal-argoai — APPROVED** [review]
Left some comments. Let me know what you think.

**Review by tagarwal-argoai — COMMENTED** [review]
Left some comments. Let me know what you think.

**Review by tagarwal-argoai — COMMENTED** [review]
Left some comments. Let me know what you think.

File: demo_usage/visualize_30hz_benchmark_data_on_map.py [hunk 11]
```diff
@@ -327,7 +327,7 @@ def render_front_camera_on_axis(self, ax: plt.Axes, timestamp: int, log_id: str)
 
 
 def visualize_30hz_benchmark_data_on_map(args: Any) -> None:
-    """ """
+    """Function to render lidar, imagery, and map elements overlaid on both a bird's-eye-view and egoview."""
```
Comment (johnwlambert) [inline demo_usage/visualize_30hz_benchmark_data_on_map.py:330]: Thanks for the updates Ben -- should actually be `"Function to render lidar, tracks, and map elements overlaid on a bird's-eye-view"`

Comment (benjaminrwilson) [reply demo_usage/visualize_30hz_benchmark_data_on_map.py]: Resolved.

File: tests/test_ffmpeg_utils_unit.py [hunk 12]
```diff
@@ -5,14 +5,14 @@
...
-    """ """
+    """Test writing sequential video with `ffmpeg`."""
     image_prefix = "imgs_%d.jpg"
     output_prefix = "out"
     write_video(image_prefix, output_prefix)
 
 
 def test_ffmpeg_nonseq_frame_vid_smokescreen() -> None:
-    """ """
+    """Test writing non-sequential video with `ffmpeg`."""
```
Comment (johnwlambert) [inline tests/test_ffmpeg_utils_unit.py:15]: it actually is a sequential video, just the frames arent consecutive integers. More like they are ordered nanosecond timestamps. This requires different syntax in ffmpeg

Comment (benjaminrwilson) [reply tests/test_ffmpeg_utils_unit.py]: Updated.

**Review by johnwlambert — APPROVED** [review]
LGTM. Feel free to merge if CI passes

**Review by tagarwal-argoai — APPROVED** [review]
LGTM!

**Comment by johnwlambert** [comment]
@jagjeet-singh could you merge this if it looks OK to you?

**Pull request merged**